lib/modules/datasource/pypi/index.ts

```diff
@@ -34,45 +34,50 @@ export class PypiDatasource extends Datasource {
     packageName,
     registryUrl,
   }: GetReleasesConfig): Promise<ReleaseResult | null> {
-    let dependency: ReleaseResult | null = null;
     // TODO: null check (#22198)
-    const hostUrl = ensureTrailingSlash(
-      registryUrl!.replace('https://pypi.org/simple', 'https://pypi.org/pypi'),
-    );
+    const hostUrl = ensureTrailingSlash(registryUrl!);
     const normalizedLookupName = PypiDatasource.normalizeName(packageName);
 
-    // not all simple indexes use this identifier, but most do
-    if (hostUrl.endsWith('/simple/') || hostUrl.endsWith('/+simple/')) {
+    const simpleHostUrl = hostUrl.replace(
+      'https://pypi.org/pypi',
+      'https://pypi.org/simple',
+    );
+    const simpleDependencies = await this.getResultsViaSimple(
+      normalizedLookupName,
+      simpleHostUrl,
+    ).catch((err) => {
+      if (err.statusCode !== 404) {
+        throw err;
+      }
       logger.trace(
         { packageName, hostUrl },
-        'Looking up pypi simple dependency',
+        'Simple api not found. Looking up pypijson api as fallback.',
       );
-      dependency = await this.getSimpleDependency(
-        normalizedLookupName,
-        hostUrl,
-      );
-    } else {
-      logger.trace({ packageName, hostUrl }, 'Looking up pypi api dependency');
-      try {
-        // we need to resolve early here so we can catch any 404s and fallback to a simple lookup
-        dependency = await this.getDependency(normalizedLookupName, hostUrl);
-      } catch (err) {
-        if (err.statusCode !== 404) {
-          throw err;
-        }
-
-        // error contacting json-style api -- attempt to fallback to a simple-style api
-        logger.trace(
-          { packageName, hostUrl },
-          'Looking up pypi simple dependency via fallback',
-        );
-        dependency = await this.getSimpleDependency(
-          normalizedLookupName,
-          hostUrl,
-        );
+      return null;
+    });
+    logger.trace({ packageName, hostUrl }, 'Querying json api for metadata');
+    const pypiJsonHostUrl = hostUrl.replace(
+      'https://pypi.org/simple',
+      'https://pypi.org/pypi',
+    );
+    const pypiJsonDependencies = await this.getResultsViaPyPiJson(
+      normalizedLookupName,
+      pypiJsonHostUrl,
+    ).catch((err) => {
+      if (simpleDependencies === null) {
+        throw err;
       }
+      logger.trace(
+        { packageName, hostUrl },
+        'Json api lookup failed but got simple results.',
+      );
+      return null;
+    });
+    if (simpleDependencies === null && pypiJsonDependencies === null) {
+      return null;
     }
-    return dependency;
+    // merge results
+    return Object.assign({}, simpleDependencies, pypiJsonDependencies);
```

Comment:
```suggestion
    return {
      ...simpleDependencies,
      ...pypiJsonDependencies
    };
```
Will all pypiJsonDependencies the same information as the simple ones. 
As this will overwrite `releases`.

Reply:
You are right. Maybe concatenating the found releases is better? If I saw it correctly, the dependencies should be de-duplicated later